src/modbus_proxy.py

```diff
@@ -171,13 +172,30 @@ async def _write_read(self, data):
         await self._write(data)
         return await self._read()
 
+    def _transform_request(self, request):
+        request_unit_id = request[6]
+
+        if self.unit_id_remapping is None:
+            return request
+
+        if self.unit_id_remapping[request_unit_id] is None:
```

Comment:
This will fail in line 181 if the incoming `request_unit_id` is not in the map

Reply:
Good catch. I rarely program in Python and mistakenly tested incorrectly whilst writing this line, thinking that the key access would always return `None` if the key doesn't exist (it doesn't).